backend/src/main/java/com/quizzard/quizzard/service/QuizAttemptService.java

```diff
@@ -98,21 +104,89 @@ public QuizAttemptResponse updateQuizAttempt(String jwtToken, Long id, Map<Strin
         }
         if (quizAttemptUpdateRequest.get("completed") == null)
             throw new RuntimeException("completed field is required.");
-        else{
+        else {
             if (quizAttempt.getIsCompleted())
                 throw new RuntimeException("Quiz attempt is already completed.");
-            else if ((boolean) quizAttemptUpdateRequest.get("completed")){
+            else if ((boolean) quizAttemptUpdateRequest.get("completed")) {
                 //quizAttempt.setScore(quizAttempt.getQuiz().getQuestions().size());
+                calculateQuestionPoint(jwtToken, id);
                 quizAttempt.setCompletedAt(new Date());
                 quizAttempt.setIsCompleted(true);
                 quizAttemptRepository.save(quizAttempt);
-            }
-            else {
+            } else {
                 quizAttempt.setIsCompleted(false);
                 quizAttemptRepository.save(quizAttempt);
             }
             return new QuizAttemptResponse(quizAttempt);
         }
     }
 
+
+    public void calculateQuestionPoint(String jwtToken, Long quizAttemptId) {
```

Comment:
you can directly use user in the argument field instead of jwtToken

Reply:
Done